src/test/java/bio/terra/service/snapshotbuilder/SnapshotAccessRequestModelTest.java

```diff
@@ -0,0 +1,162 @@
+                "Name: %s\nGroups:\n%s\n%s",
+                cohortName, expectedCriteriaGroupString, expectedCriteriaGroupString)));
+  }
+
+  private SnapshotBuilderProgramDataRangeCriteria generateRangeCriteria() {
+    SnapshotBuilderProgramDataRangeCriteria rangeCriteria =
+        new SnapshotBuilderProgramDataRangeCriteria();
+    rangeCriteria
+        .low(1960)
+        .high(1980)
+        .id(SnapshotBuilderTestData.YEAR_OF_BIRTH_PROGRAM_DATA_ID)
+        .kind(SnapshotBuilderCriteria.KindEnum.RANGE);
+    return rangeCriteria;
+  }
+
+  private SnapshotBuilderProgramDataListCriteria generateListCriteria() {
+    SnapshotBuilderProgramDataListCriteria listCriteria =
+        new SnapshotBuilderProgramDataListCriteria();
+    listCriteria
+        .values(List.of(0, 1, 2))
+        .id(SnapshotBuilderTestData.RACE_PROGRAM_DATA_ID)
+        .kind(SnapshotBuilderCriteria.KindEnum.LIST);
+    return listCriteria;
+  }
+
+  private SnapshotBuilderDomainCriteria generateDomainCriteria() {
+    SnapshotBuilderDomainCriteria domainCriteria = new SnapshotBuilderDomainCriteria();
+    domainCriteria
+        .conceptId(401)
+        .id(SnapshotBuilderTestData.CONDITION_OCCURRENCE_DOMAIN_ID)
+        .kind(SnapshotBuilderCriteria.KindEnum.DOMAIN);
+    return domainCriteria;
+  }
+
+  private SnapshotAccessRequestModel generateSnapshotAccessRequestModel() {
+    return new SnapshotAccessRequestModel(
+        UUID.randomUUID(),
+        "snapshot name",
+        "snapshot research purpose",
+        UUID.randomUUID(),
+        SnapshotBuilderTestData.createSnapshotBuilderRequest(),
+        "a@b.com",
+        Instant.now(),
+        Instant.now(),
+        SnapshotAccessRequestStatus.SUBMITTED,
+        UUID.randomUUID(),
+        "flightid");
+  }
+
+  private void compareModelAndResponseFields(
+      SnapshotAccessRequestModel model, SnapshotAccessRequestResponse response) {
+    String expectedSummaryString =
+        "Participants included:\nName: cohort\nGroups:\nMust meet all of:\nThe following concepts from Race: \nCondition Concept Id: 100\nYear of birth between 1950 and 2000\nTables included:Drug, Condition\n";
+
+    assertThat(model.id(), is(response.getId()));
+    assertThat(model.sourceSnapshotId(), is(response.getSourceSnapshotId()));
+    assertThat(model.snapshotResearchPurpose(), is(response.getSnapshotResearchPurpose()));
+    assertThat(model.snapshotSpecification(), is(response.getSnapshotSpecification()));
+    assertThat(model.createdBy(), is(response.getCreatedBy()));
+    assertThat(model.createdDate().toString(), is(response.getCreatedDate()));
+    assertThat(model.statusUpdatedDate().toString(), is(response.getStatusUpdatedDate()));
+    assertThat(model.status(), is(response.getStatus()));
+    assertThat(model.flightid(), is(response.getFlightid()));
+    assertThat(model.createdSnapshotId(), is(response.getCreatedSnapshotId()));
+    assertThat(response.getSummary(), is(expectedSummaryString));
```

Comment:
this could be a good place to use org.hamcrest.Matchers.equalToCompressingWhiteSpace or something like
```
  public static void assertQueryEquals(String expectedSql, String actualSql) {
    assertEquals(stripSpaces(expectedSql), stripSpaces(actualSql), "The sql generated is correct");
  }
```
which is defined in CriteriaQueryBuilderTest